controllers/grid/admin/languages/AdminLanguageGridHandler.inc.php

```diff
@@ -416,47 +417,33 @@ function setPrimaryLocale($args, $request) {
 	//
 	// Helper methods.
 	//
-	/**
-	 * Update the locale support state (enabled or disabled).
-	 * @param $request Request
-	 * @param $rowId string The locale row id.
-	 * @param $enable boolean Enable locale flag.
-	 */
-	function _updateLocaleSupportState($request, $rowId, $enable) {
-		$newSupportedLocales = array();
-		$gridData = $this->getGridDataElements($request);
-
-		foreach ($gridData as $locale => $data) {
-			if ($data['supported']) {
-				array_push($newSupportedLocales, $locale);
-			}
-		}
-
-		if (AppLocale::isLocaleValid($rowId)) {
-			if ($enable) {
-				array_push($newSupportedLocales, $rowId);
-			} else {
-				$key = array_search($rowId, $newSupportedLocales);
-				if ($key !== false) unset($newSupportedLocales[$key]);
-			}
-		}
-
-		$site = $request->getSite();
-		$site->setSupportedLocales($newSupportedLocales);
-
-		$siteDao = DAORegistry::getDAO('SiteDAO');
-		$siteDao->updateObject($site);
-
-		$this->_updateContextLocaleSettings($request);
-	}
-
 	/**
 	 * Helper function to update locale settings in all
-	 * installed contexts, based on site locale settings.
+	 * installed journals, based on site locale settings.
```

Comment:
hmmm.. I believe the context should stay here? -- it is in the pkp-lib

Reply:
Yes, correcting. (This method was copied over from the OJS repo.)